src/common/cvt.cpp

```diff
@@ -1033,6 +1047,841 @@ void CVT_string_to_datetime(const dsc* desc,
+			}
+			break;
+
+		case 'W':
+			date_type_check(ExpectedDateType::Date, desc, pattern, cb);
+
+			if (pattern == "W")
+			{
+				int week = (times.tm_mday - 1) / 7;
+				patternResult.printf("%d", week + 1);
+			}
+			else if (pattern == "WW")
+			{
+				int week = NoThrowTimeStamp::convertGregorianDateToWeekDate(times);
+				patternResult.printf("%d", week);
+			}
+			break;
+
+		case 'D':
+			date_type_check(ExpectedDateType::Date, desc, pattern, cb);
+
+			if (pattern == "D")
+				patternResult.printf("%d", times.tm_wday + 1);
+			else if (pattern == "DAY")
+				patternResult.printf("%s", FB_LONG_DAYS_UPPER[times.tm_wday]);
+			else if (pattern == "DD")
+				patternResult.printf("%d", times.tm_mday);
+			else if (pattern == "DDD")
+				patternResult.printf("%d", times.tm_yday + 1);
+			else if (pattern == "DY")
+				patternResult.printf("%s", FB_SHORT_DAYS[times.tm_wday]);
+			break;
+
+		case 'J':
+			date_type_check(ExpectedDateType::Date, desc, pattern, cb);
+
+			if (pattern == "J")
+			{
+				int JulianDay = NoThrowTimeStamp::convertGregorianDateToJulianDate(times.tm_year + 1900, times.tm_mon + 1,
+					times.tm_mday);
+
+				patternResult.printf("%d", JulianDay);
+			}
+			break;
+
+		case 'H':
+			date_type_check(ExpectedDateType::Time, desc, pattern, cb);
+
+			if (pattern == "HH"||
+				pattern == "HH12")
+			{
+				const char* period;
+				int hours = times.tm_hour;
+
+				if (hours >= 12)
+				{
+					period = "PM";
+					if (hours > 12)
+						hours -= 12;
+				}
+				else
+				{
+					period = "AM";
+					if (hours == 0)
+						hours = 12;
+				}
+
+				patternResult.printf("%d %s", hours, period);
+			}
+			else if (pattern == "HH24")
+				patternResult.printf("%d", times.tm_hour);
+			break;
+
+		case 'S':
+			date_type_check(ExpectedDateType::Time, desc, pattern, cb);
+
+			if (pattern == "SS")
+				patternResult.printf("%d", times.tm_sec);
+			else if (pattern == "SSSSS")
+			{
+				int secondsInDay = times.tm_hour * 60 * 60 + times.tm_min * 60 + times.tm_sec;
+				patternResult.printf("%d", secondsInDay);
+			}
+			break;
+
+		case 'F':
+			date_type_check(ExpectedDateType::Time, desc, pattern, cb);
+
+			if (!strncmp(pattern.data(), "FF", pattern.length() - 1))
+			{
+				int number = pattern.back() - '0';
+				if (number < 1 || number > 9)
+				{
+					invalidPatternException(pattern, cb);
+				}
+
+				const int fractionsPrecision = fractions != 0 ? std::log10(fractions) + 1 : 1;
+				int additionalZerosCount = 0;
+				if (number > fractionsPrecision)
+				{
+					additionalZerosCount = number - fractionsPrecision;
+					number = fractionsPrecision;
+				}
+
+				patternResult.printf("%d%.*s", fractions / (int) powf(10, fractionsPrecision - number),
+					additionalZerosCount, "00000000");
+			}
+			break;
+
+		case 'T':
+			date_type_check(ExpectedDateType::TimeZone, desc, pattern, cb);
+
+			if (pattern == "TZH")
+			{
+				if (previousPattern == "TZM")
+					timezoneOffset *= sign(timezoneOffset);
```

Comment:
Please explain this line.

Reply:
I do this to remove the offset sign, so if we have the format "TZH:TZM", our output looks like "-03:30". But maybe I should use abs() for readability.